app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -355,6 +356,226 @@ public function uploadGeoJSONFile(Request $request)
+                          'nonSurpassEstimatedArea' => $isValidArea,
+                          'completeData' => SitePolygonValidator::isValid('SCHEMA', $validationGeojson) && SitePolygonValidator::isValid('DATA', $validationGeojson),
+                        ];
+
+                        $approvalValidations = array_filter($validations, function ($key) {
+                            return ! in_array($key, ['nonSurpassEstimatedArea', 'completeData']);
+                        }, ARRAY_FILTER_USE_KEY);
+
+                        $canBeApproved = array_reduce($approvalValidations, fn ($carry, $item) => $carry && $item, true);
+
+                        $csvData[] = $this->makeCSVRow($feature, $validations, $canBeApproved, false);
+                    } else {
+                        $csvData[] = $this->makeCSVRow($feature, [], false, true);
+                    }
+                }
+            }
+        }
+
+        return $csvData;
+    }
+
+    public function makeCSVRow($feature, $validations, $canBeApproved, $isEmpty)
+    {
+        if ($isEmpty) {
+            return [
+              'polygon_name' => $feature['properties']['poly_name'] ?? 'Unnamed Polygon',
+              'site_uuid' => '',
+              'No Overlapping' => '',
+              'No Self-intersection' => '',
+              'Inside Coordinate System' => '',
+              'Inside Size Limit' => '',
+              'Within Country' => '',
+              'No Spikes' => '',
+              'Polygon Type' => '',
+              'Within Total Area Expected' => '',
+              'Completed Data' => '',
+              'Can Be Approved?' => 'No Site ID Available',
+            ];
+        } else {
+            return [
+              'polygon_name' => $feature['properties']['poly_name'] ?? 'Unnamed Polygon',
+              'site_uuid' => $feature['properties']['site_id'],
+              'No Overlapping' => $validations['nonOverlapping'] ? 'TRUE' : 'FALSE',
+              'No Self-intersection' => $validations['nonSelfIntersection'] ? 'TRUE' : 'FALSE',
+              'Inside Coordinate System' => $validations['insideCoordinateSystem'] ? 'TRUE' : 'FALSE',
+              'Inside Size Limit' => $validations['nonSurpassSizeLimit'] ? 'TRUE' : 'FALSE',
+              'Within Country' => $validations['insideCountry'] ? 'TRUE' : 'FALSE',
+              'No Spikes' => $validations['noSpikes'] ? 'TRUE' : 'FALSE',
+              'Polygon Type' => $validations['validPolygonType'] ? 'TRUE' : 'FALSE',
+              'Within Total Area Expected' => $validations['nonSurpassEstimatedArea'] ? 'TRUE' : 'FALSE',
+              'Completed Data' => $validations['completeData'] ? 'TRUE' : 'FALSE',
+              'Can Be Approved?' => $canBeApproved ? 'YES' : 'NO',
+            ];
+        }
+    }
+
+    public function uploadGeoJSONFileWithValidation(Request $request)
+    {
+        ini_set('max_execution_time', '-1');
+        ini_set('memory_limit', '-1');
```

Comment:
Yes, the endpoint is slow. I changed to 4 mins, so it is similar to nginx timeout.
This Controller is slow, we are pending on limiting size for uploads.

Reply:
I'll approve for now, but this will have to change in a future ticket. Anything longer than about 15 seconds should not be handled in a synchronous API action.